### PR TITLE
[CUSTENG-3584] Add support for refreshing message center messages in flutter

### PR DESCRIPTION
**Pull request by devinsmythe**
### What do these changes do?

These changes should allow customers using the flutter SDK to manually refresh the inbox. Currently refreshes only happen when opening the app or when a push is received with a message center message.

### Why are these changes necessary?

A customer wants to be able to refresh the inbox inside the app.
https://urbanairship.atlassian.net/browse/CUSTENG-3584

### How did you verify these changes?

I tested the changes on an android device.
**I have not tested the changes on iOS**, I could use some help testing my iOS changes out!

### Anything else a reviewer should know?

iOS still needs to be tested. I'm not 100% sure I called the correct iOS method.

### CODE REVIEW

File: android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt [hunk 1]
```diff
@@ -202,6 +203,11 @@ class AirshipPlugin : MethodCallHandler, FlutterPlugin {
         result.success(messages)
     }
 
+    private fun fetchMessages(result: Result) {
+        MessageCenter.shared().inbox.fetchMessages()
```
Comment (rlepinski) [inline android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt:207]: wait to set the result until you get a callback:
```
MessageCenter.shared().inbox.fetchMessages { success -> 
  result.success(success)            
}

File: ios/Classes/SwiftAirshipPlugin.swift [hunk 2]
```diff
@@ -392,6 +392,10 @@ UADeepLinkDelegate, UAPushNotificationDelegate {
         }
     }
 
+    private func fetchMessages(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        UAMessageCenter.shared().messageList.refreshInboxWithCompletionHandler()
```
Comment (rlepinski) [inline ios/Classes/SwiftAirshipPlugin.swift:396]: need to set the result as well, ideally in the completion handler

File: lib/src/airship_flutter.dart [hunk 3]
```diff
@@ -250,6 +250,10 @@ class Airship {
     return await _channel.invokeMethod('deleteInboxMessage', message.messageId);
   }
 
+  static Future<void> refreshInbox() async {
```
Comment (rlepinski) [inline lib/src/airship_flutter.dart:253]: can be updated to Future<bool> if you do the chaining right with the completion handlers

File: lib/src/airship_flutter.dart [hunk 4]
```diff
@@ -250,6 +250,10 @@ class Airship {
...
+    return await _channel.invokeMethod("fetchMessages");
```
Comment (rlepinski) [inline lib/src/airship_flutter.dart:254]: you should keep method names the same so its easier to trace, `refreshInbox` here and in the native layer

**Comment by rlepinski** [comment]
Ill look into the failing CI.

The suggestions above will make it easier to implement a refresh loading indicator in the app since they will be notified with the refresh is finished using the future.

**Comment by devinsmythe** [comment]
@rlepinski having issues with the swift implementation. Can you take a look at what I'm doing wrong?

**Comment by devinsmythe** [comment]
Hopefully I got the future stuff working correctly. I tested it on android with the sample and I was able to retrieve the future value doing this so seems like it's working:
`Airship.refreshInbox().then((value) => print("Future value: " + value.toString()));`

**Review by oristanovic — APPROVED** [review]

**Review by Apekka — APPROVED** [review]

**Pull request merged**